File: Codes in Python3/Dynamic Programming/Frog with K Jumps.py
from typing import *
from math import *
def minimizeCost(n : int, k : int, heights : List[int]) -> int:
    dp=[0]*n
    for i in range(1,n):
        minE=inf
        for j in range(1,k+1):
            if i-j>=0:
                minE=min(minE,dp[i-j]+abs(heights[i]-heights[i-j]))
        dp[i]=minE
    return dp[n-1]
    pass

File: Codes in C++/Dynamic Programming/Frog with K Jumps.cpp
int minimizeCost(int n, int k, vector<int> &height){
    vector<int> dp(n,0);
    dp[0]=0;
    for(int i=1;i<n;i++)
    {   int minE=INT_MAX;
        for(int j=1;j<=k;j++)
        {
            if(i-j>=0)
            {
                minE=min(minE,dp[i-j]+abs(height[i]-height[i-j]));
            }
        }
        dp[i]=minE;
    }
    return dp[n-1];
}